spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/pubsub/GcpPubSubAutoConfiguration.java

```diff
@@ -383,8 +405,14 @@ public SubscriptionAdminClient subscriptionAdminClient(
 	@ConditionalOnMissingBean
 	public TransportChannelProvider transportChannelProvider() {
 		return InstantiatingGrpcChannelProvider.newBuilder()
-			.setKeepAliveTime(Duration.ofMinutes(this.gcpPubSubProperties.getKeepAliveIntervalMinutes()))
-			.build();
+				.setKeepAliveTime(Duration.ofMinutes(this.gcpPubSubProperties.getKeepAliveIntervalMinutes()))
+				.build();
+	}
+
+	@Bean
+	@ConditionalOnClass(HealthIndicator.class)
```

Comment:
The `@ConditionalOnClass` cannot be on method, at least not in its `.class` form.
Anyway when we load a class, all its methods are parsed and if there is no super class on the classpath, we are going to fail with `NoClassDefFoundError`.

If you really depend on some class it is recommended to place your bean definition into a separate (inner, protected) `@Configuration` class with that `@ConditionalOnClass`. This way all the methods in the class won't be parsed and we won't fail with the mentioned error.

I'm not sure if there is such a note in Spring Boot docs, but at least that is my experience and that's why we have that complexity for nested configuration classes in the out-of-the-box auto-configuration.

On the other hand: why there is no `@ConditionalOnBean(PubSubHealthIndicator.class)` ?
Don't you expect that end-user can override a built-in indicator?

Reply:
My bad, there was an initial commit where I had the @ConditionalOnMisingBean, and I just messed this with the @ConditionalOnClass, thanks for pointing this out @artembilan